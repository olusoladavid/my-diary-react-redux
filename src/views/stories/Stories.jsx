import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import StoryCard from "./StoryCard.jsx";
import { fetchStories } from "../../actions/stories";

export class Stories extends React.Component {
  state = {
    currentView: "My Stories"
  };

  componentDidMount() {
    const { fetchStories } = this.props;
    fetchStories();
  }

  render() {
    const { storyList, loading } = this.props;
    const { currentView } = this.state;
    const emptyStory = {
      title: "You have not added any story",
      content: "Click on this card to add your first story",
      created_on: null,
      id: null
    };
    return (
      <React.Fragment>
        <div>
          <h1 className="story-list__title js-story-list-title">{currentView}</h1>
          <div className="story-list js-story-list">
            {storyList.length && !loading ? (
              storyList.map(story => <StoryCard key={story.id} story={story} />)
            ) : (
              <StoryCard story={emptyStory} link="/new-story" />
            )}
          </div>
          <div className="story-list__load-btn">
            <span href="#" className="button button--success pagination-btn js-stories-prev" data-viewable="true">
              <i className="fas fa-arrow-left" /> Prev{" "}
            </span>
            <span href="#" className="button button--success pagination-btn js-stories-more" data-viewable="true">
              More <i className="fas fa-arrow-right" />
            </span>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

Stories.propTypes = {
  fetchStories: PropTypes.func,
  storyList: PropTypes.array,
  loading: PropTypes.bool
};

const mapStateToProps = state => ({
  storyList: state.stories.storyList,
  loading: state.stories.loading
});

const mapDispatchToProps = {
  fetchStories
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Stories);
