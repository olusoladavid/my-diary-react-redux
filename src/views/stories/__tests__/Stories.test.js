import { Stories } from "../Stories";

const props1 = {
  storyList: [],
  fetchStories: jest.fn(),
  loading: false
};

const props2 = {
  ...props1,
  storyList: [{}]
};

it("renders without crashing", () => {
  const wrapper = shallow(<Stories {...props1} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  const wrapper = shallow(<Stories {...props2} />);
  expect(wrapper).toMatchSnapshot();
});
