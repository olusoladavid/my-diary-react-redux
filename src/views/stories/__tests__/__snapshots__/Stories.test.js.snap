// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <div>
    <h1
      className="story-list__title js-story-list-title"
    >
      My Stories
    </h1>
    <div
      className="story-list js-story-list"
    >
      <StoryCard
        link="/new-story"
        story={
          Object {
            "content": "Click on this card to add your first story",
            "created_on": null,
            "id": null,
            "title": "You have not added any story",
          }
        }
      />
    </div>
    <div
      className="story-list__load-btn"
    >
      <span
        className="button button--success pagination-btn js-stories-prev"
        data-viewable="true"
        href="#"
      >
        <i
          className="fas fa-arrow-left"
        />
         Prev
         
      </span>
      <span
        className="button button--success pagination-btn js-stories-more"
        data-viewable="true"
        href="#"
      >
        More 
        <i
          className="fas fa-arrow-right"
        />
      </span>
    </div>
  </div>
</Fragment>
`;

exports[`renders without crashing 2`] = `
<Fragment>
  <div>
    <h1
      className="story-list__title js-story-list-title"
    >
      My Stories
    </h1>
    <div
      className="story-list js-story-list"
    >
      <StoryCard
        story={Object {}}
      />
    </div>
    <div
      className="story-list__load-btn"
    >
      <span
        className="button button--success pagination-btn js-stories-prev"
        data-viewable="true"
        href="#"
      >
        <i
          className="fas fa-arrow-left"
        />
         Prev
         
      </span>
      <span
        className="button button--success pagination-btn js-stories-more"
        data-viewable="true"
        href="#"
      >
        More 
        <i
          className="fas fa-arrow-right"
        />
      </span>
    </div>
  </div>
</Fragment>
`;
